Core/Core/Common/CommonModels/AppEnvironment/AppEnvironmentOverride.swift

```diff
@@ -127,6 +127,26 @@ extension AppEnvironment {
 
 // MARK: - Utils
 
+public struct RoutePath {
+    let value: String
+    let env: AppEnvironment
+
+    /// This method returns the correct path using env `baseURL`
+    /// if it was different from the one used on `AppEnvironment.shared.api`
+    func correctedPath() -> String {
```

Comment:
Maybe a name called `asRoute` would be better since this method can return either a fully qualified URL or just a url path but its name suggests that it always return a path.